### PR TITLE
CATTY-456 Convert Sound to Swift

### PR DESCRIPTION
**Pull request by dwildr**
The sound data model was converted to Swift and refactored and all the usages of the Sound class in the code where updated.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by zwerg4 — APPROVED** [review]

File: src/Catty/DataModel/Sound/Sound.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        self.name = name
+        self.fileName = fileName
+        self.playing = false
+        super.init()
+    }
+
+    @objc(pathForScene:)
+    func path(for scene: Scene) -> String? {
+        guard let soundsPath = scene.soundsPath() else {
+            return String(format: "/%@", self.fileName)
```
Comment (m-herold) [inline src/Catty/DataModel/Sound/Sound.swift:40]: Please return `nil` if the `soundsPath` does not exist/is not set.

File: src/Catty/Parser&Serialializer/Old Parser/ProjectParser.m [hunk 2]
```diff
@@ -143,6 +142,13 @@ - (id)parseNode:(GDataXMLElement*)node withParent:(XMLObjectReference*)parent
         if (lookName && lookFileName) {
             object = [[Look alloc] initWithName:[lookName stringValue] andPath:[lookFileName stringValue]];
         }
+    } else if ([className isEqualToString:@"Sound"]) {
```
Comment (m-herold) [inline src/Catty/Parser&Serialializer/Old Parser/ProjectParser.m:145]: Nice, thank you!

**Review by m-herold — CHANGES_REQUESTED** [review]
Thanks for your PR. I have left one minor suggestion, otherwise everything is looking good.

File: src/Catty/DataModel/Sound/Sound.swift (hunk 1 above)
Comment (dwildr) [reply src/Catty/DataModel/Sound/Sound.swift]: Done 👌🏼

**Review by m-herold — APPROVED** [review]
Thanks!

**Pull request merged**